larray/io/session.py

```diff
@@ -67,6 +69,15 @@ def close(self):
         """
         raise NotImplementedError()
 
+    def _create_temporay_file(self):
```

Comment:
rename to _get_temporary_fname and call it from `__init__` instead? Also I think it would be better to modify self.fname directly and store the original (unmodified) fname in some other attribue (eg self.final_fname), so that each backend do not have to worry about tmp file in their open_for_write method.